tests/components/hikvision/test_config_flow.py

```diff
@@ -0,0 +1,348 @@
+"""Test the Hikvision config flow."""
+
+from unittest.mock import AsyncMock, MagicMock
+
+import requests
+
+from homeassistant.components.hikvision.const import DOMAIN
+from homeassistant.config_entries import SOURCE_IMPORT, SOURCE_USER
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_USERNAME,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from .conftest import (
+    TEST_DEVICE_ID,
+    TEST_DEVICE_NAME,
+    TEST_HOST,
+    TEST_PASSWORD,
+    TEST_PORT,
+    TEST_USERNAME,
+)
+
+from tests.common import MockConfigEntry
+
+
+async def test_form(
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+    mock_hikcamera_config_flow: MagicMock,
+) -> None:
+    """Test we get the form and can create entry."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            CONF_HOST: TEST_HOST,
+            CONF_PORT: TEST_PORT,
+            CONF_USERNAME: TEST_USERNAME,
+            CONF_PASSWORD: TEST_PASSWORD,
+            CONF_SSL: False,
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert result2["type"] is FlowResultType.CREATE_ENTRY
+    assert result2["title"] == TEST_DEVICE_NAME
+    assert result2["data"] == {
+        CONF_HOST: TEST_HOST,
+        CONF_PORT: TEST_PORT,
+        CONF_USERNAME: TEST_USERNAME,
+        CONF_PASSWORD: TEST_PASSWORD,
+        CONF_SSL: False,
+    }
+
+
+async def test_form_with_ssl(
```

Comment:
1. Can the interface be hosted on a port different than 80 or 443?
2. Why do we ask both the port and the ssl?

Reply:
1. I don't know. For my NVR it can't be but maybe because the machine is 10 years old. The functionality was there before so I think I'm just leaving it.
2. I assume since you can run on a non-standard port you can choose to use that port with SSL or not.